### PR TITLE
More details in `cuts.describe()` + fix for `trim_to_unsupervised_segments()`

### PR DESCRIPTION
**Pull request by desh2608**
1. Added an optional `full:bool=False` parameter in `CutSet.describe()`. If set to True, it prints additional information about the cut set, such as the following for AMI:

```
Cut statistics:
╒═══════════════════════════╤══════════╕
│ Cuts count:               │ 133      │
├───────────────────────────┼──────────┤
│ Total duration (hh:mm:ss) │ 79:23:03 │
├───────────────────────────┼──────────┤
│ mean                      │ 2148.7   │
├───────────────────────────┼──────────┤
│ std                       │ 867.4    │
├───────────────────────────┼──────────┤
│ min                       │ 477.9    │
├───────────────────────────┼──────────┤
│ 25%                       │ 1509.8   │
├───────────────────────────┼──────────┤
│ 50%                       │ 2181.7   │
├───────────────────────────┼──────────┤
│ 75%                       │ 2439.9   │
├───────────────────────────┼──────────┤
│ 99%                       │ 5300.7   │
├───────────────────────────┼──────────┤
│ 99.5%                     │ 5355.3   │
├───────────────────────────┼──────────┤
│ 99.9%                     │ 5403.2   │
├───────────────────────────┼──────────┤
│ max                       │ 5415.2   │
├───────────────────────────┼──────────┤
│ Recordings available:     │ 133      │
├───────────────────────────┼──────────┤
│ Features available:       │ 0        │
├───────────────────────────┼──────────┤
│ Supervisions available:   │ 102222   │
╘═══════════════════════════╧══════════╛
Speech duration statistics:
╒══════════════════════════════╤══════════╤═══════════════════════════╕
│ Total speech duration        │ 64:59:51 │ 81.88% of recording       │
├──────────────────────────────┼──────────┼───────────────────────────┤
│ Total speaking time duration │ 74:33:09 │ 93.91% of recording       │
├──────────────────────────────┼──────────┼───────────────────────────┤
│ Total silence duration       │ 14:23:12 │ 18.12% of recording       │
├──────────────────────────────┼──────────┼───────────────────────────┤
│ Single-speaker duration      │ 56:18:24 │ 70.93% (86.63% of speech) │
├──────────────────────────────┼──────────┼───────────────────────────┤
│ Overlapped speech duration   │ 08:41:28 │ 10.95% (13.37% of speech) │
╘══════════════════════════════╧══════════╧═══════════════════════════╛
Speech duration statistics by number of speakers:
╒══════════════════════╤═══════════════════════╤════════════════════════════╤═══════════════╤══════════════════════╕
│ Number of speakers   │ Duration (hh:mm:ss)   │ Speaking time (hh:mm:ss)   │ % of speech   │ % of speaking time   │
╞══════════════════════╪═══════════════════════╪════════════════════════════╪═══════════════╪══════════════════════╡
│ 1                    │ 56:18:24              │ 56:18:24                   │ 86.63%        │ 75.53%               │
├──────────────────────┼───────────────────────┼────────────────────────────┼───────────────┼──────────────────────┤
│ 2                    │ 07:51:44              │ 15:43:28                   │ 12.10%        │ 21.09%               │
├──────────────────────┼───────────────────────┼────────────────────────────┼───────────────┼──────────────────────┤
│ 3                    │ 00:47:36              │ 02:22:47                   │ 1.22%         │ 3.19%                │
├──────────────────────┼───────────────────────┼────────────────────────────┼───────────────┼──────────────────────┤
│ 4                    │ 00:02:08              │ 00:08:31                   │ 0.05%         │ 0.19%                │
├──────────────────────┼───────────────────────┼────────────────────────────┼───────────────┼──────────────────────┤
│ Total                │ 64:59:51              │ 74:33:09                   │ 100.00%       │ 100.00%              │
╘══════════════════════╧═══════════════════════╧════════════════════════════╧═══════════════╧══════════════════════╛
```
In general, we differentiate "speech time" from "speaking time" even when `full` is False.

2. Fix an issue in `trim_to_unsupervised_segments()`. I think the current implementation did not work when overlapping supervisions were present in the cut.

This adds a dependency on the [tabulate](https://github.com/astanin/python-tabulate) package, but it is pretty light-weight so maybe it is okay to add it?

### CODE REVIEW

File: lhotse/cut/set.py [hunk 1]
```diff
@@ -606,80 +605,268 @@ def save(cut: DataCut):
...
+            │ 2                    │ 07:51:44              │ 15:43:28                   │ 12.10%        │ 21.09%               │
+            ├──────────────────────┼───────────────────────┼────────────────────────────┼───────────────┼──────────────────────┤
+            │ 3                    │ 00:47:36              │ 02:22:47                   │ 1.22%         │ 3.19%                │
+            ├──────────────────────┼───────────────────────┼────────────────────────────┼───────────────┼──────────────────────┤
+            │ 4                    │ 00:02:08              │ 00:08:31                   │ 0.05%         │ 0.19%                │
+            ├──────────────────────┼───────────────────────┼────────────────────────────┼───────────────┼──────────────────────┤
+            │ Total                │ 64:59:51              │ 74:33:09                   │ 100.00%       │ 100.00%              │
+            ╘══════════════════════╧═══════════════════════╧════════════════════════════╧═══════════════╧══════════════════════╛
+        """
+        from tabulate import tabulate
```
Comment (csukuangfj) [inline lhotse/cut/set.py:679]: Shall we remind users about the installation in case it is not installed?

Comment (desh2608) [reply lhotse/cut/set.py:679]: Yes, good point. I will also add it in the setup.py.

**Comment by pzelasko** [comment]
Looks nice! I'm OK with merging it. BTW if we're moving into pretty displays, I was recently playing around with [rich](https://github.com/Textualize/rich) in another project and was very impressed by its capabilities. WDYT?

**Comment by desh2608** [comment]
> Looks nice! I'm OK with merging it. BTW if we're moving into pretty displays, I was recently playing around with [rich](https://github.com/Textualize/rich) in another project and was very impressed by its capabilities. WDYT?

Yes, I have in fact been using rich locally to print out Cuts and other things, since it formats the contents so well.

**Comment by pzelasko** [comment]
OK let's merge as is, maybe we'll move to rich sometime, not sure yet... no need to block this :)

**Review by pzelasko — APPROVED** [review]

**Comment by desh2608** [comment]
> OK let's merge as is, maybe we'll move to rich sometime, not sure yet... no need to block this :)

Where do you envision using rich currently?

**Comment by pzelasko** [comment]
No, let's not do this unless there is a good reason to (e.g. more tools informing about data stats/distributions/etc.)

**Pull request merged**